### PR TITLE
IGNITE-22733 Introduce TransactionChanges

### PR DESCRIPTION
**Pull request by nizhikov**
…test

Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/TransactionChanges.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+import org.apache.ignite.internal.processors.cache.KeyCacheObject;
+
+/**
+ * Changes of transaction in convenient for queries form.
+ */
+public class TransactionChanges<E> {
```
Comment (timoninmaxim) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/TransactionChanges.java:29]: Let's add javadoc for E type.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/TransactionChanges.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+     */
+    public static <R> TransactionChanges<R> empty() {
+        return (TransactionChanges<R>)EMPTY;
+    }
+
+    /**
+     * Sort new and updated entries collection.
+     * @param comparator Entries comparator
+     */
+    public void sortNewAndUpdatedEntries(Comparator<E> comparator) {
```
Comment (timoninmaxim) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/TransactionChanges.java:80]: Is it possible to specify comparator in constructor?

**Review by timoninmaxim — CHANGES_REQUESTED** [review]

File: modules/calcite/src/test/java/org/apache/ignite/internal/processors/tx/SqlTransactionsIsolationTest.java [hunk 3]
```diff
@@ -218,32 +96,29 @@ public enum ModifyApi {
...
+                for (CacheMode mode : CacheMode.values()) {
                     for (ModifyApi modify : ModifyApi.values()) {
                         for (boolean commit : new boolean[]{false, true}) {
-                            for (boolean mutli : new boolean[] {false, true}) {
+                            for (boolean mutli : new boolean[]{false, true}) {
                                 for (TransactionConcurrency txConcurrency : TransactionConcurrency.values()) {
                                     ExecutorType[] nodeExecTypes = {ExecutorType.SERVER, ExecutorType.CLIENT};
                                     ExecutorType[] thinExecTypes;
 
-                                    if (modify == SQL) {
```
Comment (timoninmaxim) [inline modules/calcite/src/test/java/org/apache/ignite/internal/processors/tx/SqlTransactionsIsolationTest.java:246]: Do we need this changes in this patch?

Comment (nizhikov) [reply modules/calcite/src/test/java/org/apache/ignite/internal/processors/tx/SqlTransactionsIsolationTest.java:246]: The main reason to include test changes in patch is to reduce main patch.
Exclude obvious changes from it.

So I thins yes, we need this :)

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/TransactionChanges.java (hunk 2 above)
Comment (nizhikov) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/TransactionChanges.java:80]: Done

**Review by timoninmaxim — APPROVED** [review]

**Pull request merged**